src/core/mount-setup.c

```diff
@@ -397,9 +397,19 @@ int mount_setup(bool loaded_policy) {
          * nspawn and the container tools work out of the box. If
          * specific setups need other settings they can reset the
          * propagation mode to private if needed. */
-        if (detect_container() <= 0)
-                if (mount(NULL, "/", NULL, MS_REC|MS_SHARED, NULL) < 0)
-                        log_warning_errno(errno, "Failed to set up the root directory for shared mount propagation: %m");
+        if (mount(NULL, "/", NULL, MS_REC|MS_SHARED, NULL) < 0)
+                log_warning_errno(errno, "Failed to set up the root directory for shared mount propagation: %m");
+
```

Comment:
If you set MS_SHARED unconditionally you ignore the comment above that says special setups can reset propagation flags, since their change will be reset again... also `detect_container()` works in this context with pid namespaces and that systemd `$container` var to do easy sandboxes setups.
Now private or unbindable mounts of the subtree will also be updated to MS_SHARED, it's not only about slaves... which from a security point of view I have to hack around it to make it integrate with systemd as pid1 inside.